src/Rules/Comparison/ImpossibleCheckTypeHelper.php

```diff
@@ -79,6 +80,18 @@ public function findSpecifiedType(
 				], true)) {
 					return null;
 				}
+				if ($functionName === 'function_exists') {
+					$functionNameType = $scope->getType($node->getArgs()[0]->value);
+
+					foreach ($functionNameType->getConstantStrings() as $constantString) {
+						$functionReflection = $this->reflectionProvider->getFunction(new Node\Name($constantString->getValue()), $scope);
+						if ($functionReflection instanceof NativeFunctionReflection) {
+							// native functions existance depends php configuration,
+							// therefore don't judge them regarding will always exist or not
+							return null;
+						}
```

Comment:
let me propose to not judge about "always exist yes/no", when `function_exists()` is called on a php-src native function.

wdyt?

---

going one step further I think it might also make sense to prevent the "always exists yes/no" error for methods, when the class involved is located in the composer vendor/ folder.

the assumption would be that the current project might support different versions of a external dependency and therefore needs to feauture detect the existance of methods.

Reply:
@ondrejmirtes do you have a opinion on the above comment?